common/persistence/tests/queue_v2_test_suite.go

```diff
@@ -481,36 +484,161 @@ func testListQueues(ctx context.Context, t *testing.T, queue persistence.QueueV2
 			NextPageToken: nil,
 		})
 		require.NoError(t, err)
-		require.Equal(t, 1, len(response.QueueNames))
-		listedQueueNames = append(listedQueueNames, response.QueueNames...)
+		require.Equal(t, 1, len(response.Queues))
+		listedQueueNames = append(listedQueueNames, response.Queues[0].QueueName)
+		require.Equal(t, int64(0), response.Queues[0].MessageCount)
 		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
 			QueueType:     queueType,
 			PageSize:      1,
 			NextPageToken: response.NextPageToken,
 		})
 		require.NoError(t, err)
-		require.Equal(t, 1, len(response.QueueNames))
-		listedQueueNames = append(listedQueueNames, response.QueueNames...)
+		require.Equal(t, 1, len(response.Queues))
+		listedQueueNames = append(listedQueueNames, response.Queues[0].QueueName)
+		require.Equal(t, int64(0), response.Queues[0].MessageCount)
 		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
 			QueueType:     queueType,
 			PageSize:      3,
 			NextPageToken: response.NextPageToken,
 		})
 		require.NoError(t, err)
-		require.Equal(t, 3, len(response.QueueNames))
-		listedQueueNames = append(listedQueueNames, response.QueueNames...)
+		require.Equal(t, 3, len(response.Queues))
+		for _, queue := range response.Queues {
+			listedQueueNames = append(listedQueueNames, queue.QueueName)
+			require.Equal(t, int64(0), queue.MessageCount)
+		}
 		response, err = queue.ListQueues(ctx, &persistence.InternalListQueuesRequest{
 			QueueType:     queueType,
 			PageSize:      1,
 			NextPageToken: response.NextPageToken,
 		})
 		require.NoError(t, err)
-		require.Equal(t, 0, len(response.QueueNames))
+		require.Equal(t, 0, len(response.Queues))
 		require.Empty(t, response.NextPageToken)
 		for _, queueName := range queueNames {
 			require.Contains(t, listedQueueNames, queueName)
+		}
+	})
+	t.Run("QueueSize", func(t *testing.T) {
```

Comment:
This is a great test, by the way. Thank you for verifying all the steps from start-to-finish because this is where a lot of bugs in DLQ v1 were

Reply:
Thank you!